src/js/MazeEntity.js

```diff
@@ -2,6 +2,18 @@ class MazeEntity {
     constructor(maze, row, col) {
         Object.assign(this, {maze, row, col});
     }
+    getNeighboringEntitiesDict(distance){
+      const data = this.maze.data, row = this.row, col = this.col;
+      return {
+        'left': col > (distance-1)? data[row][col-distance] : null,
+        'right': col < data[0].length - distance ? data[row][col+distance] : null,
+        'top':  row > (distance-1) ? data[row-distance][col] : null,
+        'bottom': row < data.length - distance ? data[row+distance][col] : null
+      }
+    }
+    getNeighboringEntitiesList(distance){
+      return Object.values(this.getNeighboringEntitiesDict(distance))
+    }
 }
 
-export default MazeEntity;
\ No newline at end of file
+export default MazeEntity;
```

Comment:
I love that we're putting more methods on this base class! (feels very validating of my choice to make one, haha.)